subsys/bluetooth/mesh/transport.c

```diff
@@ -1590,14 +1629,30 @@ void bt_mesh_heartbeat_send(void)
 }
 
 int bt_mesh_app_key_get(const struct bt_mesh_subnet *subnet, u16_t app_idx,
-			const u8_t **key, u8_t *aid)
+			u16_t addr, const u8_t **key, u8_t *aid)
 {
 	struct bt_mesh_app_key *app_key;
 
-	if (app_idx == BT_MESH_KEY_DEV) {
+	if (app_idx == BT_MESH_KEY_DEV_LOCAL ||
+	    (app_idx == BT_MESH_KEY_DEV_REMOTE &&
+	     addr == bt_mesh_primary_addr())) {
```

Comment:
Is there a rule that dev key can only be used by primary element?

Reply:
I dont think this is linked to the primary element rather the primary node address:

```
u16_t bt_mesh_primary_addr(void)
{
	return dev_primary_addr;
}
```

Reply:
> I dont think this is linked to the primary element rather the primary node address:
> 
> ```
> u16_t bt_mesh_primary_addr(void)
> {
> 	return dev_primary_addr;
> }
> ```

Isn't that the same concept? ref `Mesh Profile Spec 1.0.1 Section 2.3.4 Elements`

Reply:
Yes and no, the second paragraph gives a clue;

> The primary element is addressed using the first unicast address assigned to the node during provisioning.

Reply:
I don't see the specification stating that dev_key can `only `be used by primary element.

Reply:
I think you're right @LingoMeng, there's no such rule, it's just that all the models that currently use devkey must be in the primary element. This check should use `bt_mesh_elem_find()` instead

Reply:
Yes I think @LingoMeng is correct. This check is for the case when sending to a model which uses the receivers device key as app key and _we_ are the receiver. Right now this is the case when using the `cfg_cli` model for example.

I will change to `bt_mesh_elem_find()` instead.

Reply:
Fixed